### PR TITLE
Improve initial repository state

### LINKED ISSUE #1123: [BUG] Cannot synchronise with git repo
**Describe the bug**
Upon selecting pull from git, or synchronise, the following error is thrown:
```
An error occured during a Git operation

Attempt to read from field 'int org.eclipse.jgit.lib.AnyObjectId.w1' on a null object reference
```

**To Reproduce**
1. Initialise a pass git repo on a Linux machine
2. Create some entries (in this case, I imported my lastpass)
3. Initialise a bare git repo on a server
4. Push the pass git repo to the bare git repo
5. Import the bare git repo to the app and attempt to sync. 

**Expected behavior**
The app would pull all my passwords from my server to be available.

**Device information (please complete the following information):**
 - Device: Xiami Poco F1
 - OS: /e/ 0.11 (LineageOS llama fork (Android 9 fork))
 - App version 1.11.3

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [x] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## :scroll: Description
Make users select a GPG key ID when creating a new store and fix the default author/email to match the Git CLI.

## :bulb: Motivation and Context
We create an empty repository during init with an empty `.gpg-id` file which results in an unusable repository that
cannot be used to create passwords. This resolves that by ensuring the `.gpg-id` file contains a valid key ID. While
here, also fix the default Git author and email that were being set to empty strings which resulted in commits with the
author ` <>`. Since we do want them to remain empty by default so as to prompt the user to update them from the UI,
employ a small workaround to take correct defaults where they need to match Git.

## :green_heart: How did you test it?
Verified creating a new repository prompted me to select a key from OpenKeychain and the author was `root@localhost`.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [x] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :camera_flash: Screenshots / GIFs

<details>
<summary>Key selection fragment</summary>

screenshot-20201001-204556

</details>

### CODE REVIEW

**Comment by msfjarvis** [comment]
There is no UI cue here for the key selection part, but we might want to add one. @Skrilltrax care to pitch in?

File: app/src/main/java/com/zeapo/pwdstore/ui/onboarding/fragments/RepoLocationFragment.kt [hunk 1]
```diff
@@ -66,6 +74,26 @@ class RepoLocationFragment : Fragment(R.layout.fragment_repo_location) {
         repositoryInitAction.launch(UserPreference.createDirectorySelectionIntent(requireContext()))
     }
 
+    private val gpgKeySelectAction = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+        if (result.resultCode == AppCompatActivity.RESULT_OK) {
```
Comment (fmeum) [inline app/src/main/java/com/zeapo/pwdstore/ui/onboarding/fragments/RepoLocationFragment.kt]: This is a reasonable solution. Alternatively, we could run this flow anytime we encounter an empty (top-level?) `.gpg-id`, which could help more users get out of this situation.

Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/ui/onboarding/fragments/RepoLocationFragment.kt]: hm I can make it happen, one sec.

Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/ui/onboarding/fragments/RepoLocationFragment.kt]: This works now

**Comment by Skrilltrax** [comment]
I think we can add another fragment in the onboarding process which prompts users to select a gpg key.

**Review by fmeum — APPROVED** [review]

**Pull request merged**